junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TimeoutExtension.java

```diff
@@ -165,6 +170,26 @@ private ScheduledExecutorService getExecutor(ExtensionContext extensionContext)
 		return extensionContext.getRoot().getStore(NAMESPACE).getOrComputeIfAbsent(ExecutorResource.class).get();
 	}
 
+	private boolean isDebuggingOrDisabled(ExtensionContext extensionContext) {
+		if (extensionContext.getConfigurationParameter(TIMEOUT_MODE_PROPERTY_NAME).isPresent()) {
+			if (extensionContext.getConfigurationParameter(TIMEOUT_MODE_PROPERTY_NAME).get().equals(DISABLED)) {
+				return true;
+			}
+			else if (extensionContext.getConfigurationParameter(TIMEOUT_MODE_PROPERTY_NAME).get().equals(
+				DISABLED_ON_DEBUG)) {
+				for (final String argument : ManagementFactory.getRuntimeMXBean().getInputArguments()) {
+					if ("-Xdebug".equals(argument)) {
```

Comment:
From the [Java SE 8 documentation](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/java.html#BABHDABI) for `java` command-line options:

> *-Xdebug*
> 
>     Does nothing. Provided for backward compatibility.

So there's no need to check for `-Xdebug` in JUnit Jupiter.

Checking for the `-agentlib:jdwp` should suffice.